crates/relayer/relayer-components/src/relay/impls/event_relayers/batch_packet_event.rs

```diff
@@ -0,0 +1,187 @@
+                            .await
+                            .map_err(Relay::raise_error)?;
+                    }
+                    Ok(None) => {
+                        relay.log("no misbehaviour detected", &LevelDebug).await;
+                    }
+                    Err(e) => {
+                        relay
+                            .log(
+                                &format!("error checking for misbehaviour: {e:?}"),
+                                &LevelWarn,
+                            )
+                            .await;
+                    }
+                }
+            }
+        }
+
+        relay.relay_packets(send_packet_events.as_slice()).await?;
+
+        Ok(())
+    }
+}
+
+#[cgp_provider(BatchEventRelayerComponent)]
+impl<Relay, DstChain> BatchEventRelayer<Relay, DestinationTarget> for BatchPacketEventRelayer
+where
+    Relay: HasRelayChains<DstChain = DstChain>
+        + HasRelayClientIds
+        + CanRelayBatchAckPackets
+        + CanFilterRelayPackets
+        + CanRaiseRelayChainErrors,
+    DstChain: CanQueryChainHeight
+        + CanBuildPacketFromWriteAck<Relay::SrcChain>
+        + CanExtractFromEvent<DstChain::WriteAckEvent>,
+    MatchPacketSourceChain: RelayPacketFilter<Relay>,
+{
+    async fn relay_chain_batch_events(
+        relay: &Relay,
+        events: &[EventOf<Relay::DstChain>],
+    ) -> Result<(), Relay::Error> {
+        let dst_chain = relay.dst_chain();
+        let mut packets_info = vec![];
+        let mut batch_latest_height = dst_chain
+            .query_chain_height()
+            .await
+            .map_err(Relay::raise_error)?;
+
+        for event in events.iter() {
+            let m_ack_event = dst_chain.try_extract_from_event(PhantomData, event);
+
+            if let Some(ack_event) = m_ack_event {
+                let packet = dst_chain
+                    .build_packet_from_write_ack_event(&ack_event)
+                    .await
+                    .map_err(Relay::raise_error)?;
+
+                /*
+                   First check whether the packet is targeted for the destination chain,
+                   then use the packet filter in the relay context, as we skip `CanRelayPacket`
+                   which would have done the packet filtering.
+                */
+                if MatchPacketSourceChain::should_relay_packet(relay, &packet).await?
+                    && relay.should_relay_packet(&packet).await?
+                {
+                    let ack = dst_chain
+                        .build_ack_from_write_ack_event(&ack_event)
+                        .await
+                        .map_err(Relay::raise_error)?;
+
+                    let height = dst_chain
+                        .query_chain_height()
+                        .await
+                        .map_err(Relay::raise_error)?;
+
+                    if height > batch_latest_height {
+                        batch_latest_height = height;
+                    }
```

Comment:
why do we need this in the loop ?

isn't enough to call the `query_chain_height` at the end of the function call and use it as `batch_latest_height` ?

Reply:
Good catch, will push the fix